vuu-ui/packages/vuu-popups/src/notifications/NotificationsProvider.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useContext, useCallback, useEffect } from "react";
+import classNames from "classnames"
+import { getUniqueId } from "@finos/vuu-utils";
+
+import "./notifications.css"
+
+// animation times in milliseconds
+const toastDisplayDuration = 6000;
+const exitAnimationDuration = 1000;
+const verticalTransitionTime = 300
+
+const toastHeight = 56;
+const toastWidth = 300;
+const rightGap = 50;
+const topGap = 10;
+const leftGap = 10;
+
+const classBase = "vuuToastNotifications";
+
+export enum NotificationLevel {
+  Info = "info",
+  Success = "success",
+  Warning = "warning",
+  Error = "error",
+}
+
+type Notification = {
+  type: NotificationLevel,
+  header: string,
+  body: string,
+  id: string
+}
+
+export const NotificationsContext = React.createContext<{
+  notify: (notification: Omit<Notification, "id">) => void,
+}>({
+  notify: () => "have you forgotten to provide a NotificationProvider?"
+})
+
+export const NotificationsProvider = (props: {
+  children: JSX.Element | JSX.Element[]
+}) => {
+  const [notifications, setNotifications] = useState<Notification[]>([]);
+
+  const notify = useCallback((notification: Omit<Notification, "id">) => {
+    const newNotification = { ...notification, id: getUniqueId() }
+
+    setNotifications(prev => [...prev, newNotification])
+
+    setTimeout(() => {
+      setNotifications(prev => prev.filter(n => n !== newNotification))
+    }, toastDisplayDuration + exitAnimationDuration * 2)
```

Comment:
The fact that we're doubling `exitAnimationDuration` might make more sense if it was renamed to something like `transitionAnimationDuration`, or anything else that implies it's entrance as well as exit time.

Reply:
I have renamed it to `horizontalTransitionDuration`